posthog/warehouse/api/saved_query.py

```diff
@@ -402,6 +409,36 @@ def run(self, request: request.Request, *args, **kwargs) -> response.Response:
 
         return response.Response(status=status.HTTP_200_OK)
 
+    @action(methods=["POST"], detail=True)
+    def revert_materialization(self, request: request.Request, *args, **kwargs) -> response.Response:
+        """
+        Undo materialization, revert back to the original view.
+        (i.e. delete the materialized table and the schedule)
+        """
+        saved_query = self.get_object()
+        saved_query.sync_frequency_interval = None
+        # we still have the history preserved in our DataModelingJob
+        saved_query.last_run_at = None
+        saved_query.latest_error = None
+        saved_query.status = None
+
+        # delete the materialized table reference
+        if saved_query.table is not None:
+            saved_query.table.soft_delete()
+            saved_query.table_id = None
+
+        saved_query.save()
+
+        # delete the schedule
+        delete_saved_query_schedule(str(saved_query.id))
```

Comment:
We probably should wrap this in a try/catch and fail with a decent error message + revert the above changes. Otherwise you won't be able to revert again to delete the temporal schedule